Index: src/ru/javawebinar/basejava/MainFile.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/ru/javawebinar/basejava/MainFile.java	(date 1558300363092)
+++ src/ru/javawebinar/basejava/MainFile.java	(date 1558300363092)
@@ -0,0 +1,38 @@
+package ru.javawebinar.basejava;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+
+/**
+ * @author Vladimir Mikhaylov <vladmeh@gmail.com> on 20.05.2019.
+ * @link https://github.com/vladmeh/basejava
+ */
+
+public class MainFile {
+    public static void main(String[] args) {
+        String filePath = ".\\.gitignore";
+
+        File file = new File(filePath);
+        try {
+            System.out.println(file.getCanonicalPath());
+        } catch (IOException e) {
+            throw new RuntimeException("Error", e);
+        }
+
+        File dir = new File("./src/ru/javawebinar/basejava");
+        System.out.println(dir.isDirectory());
+        String[] list = dir.list();
+        if (list != null) {
+            for (String name : list) {
+                System.out.println(name);
+            }
+        }
+
+        try (FileInputStream fis = new FileInputStream(filePath)) {
+            System.out.println(fis.read());
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+}
